### ISSUE TITLE
Identifier 'hbs' has already been declared

### ISSUE THREAD

**Issue by HeroicEric** [ember-template-imports/ember-template-imports#22]
Ran into an issue that looks similar to what @knownasilya mentioned in https://github.com/ember-template-imports/ember-template-imports/issues/14#issuecomment-807889242

Not really sure what the cause is but I can reliably reproduce (and avoid). It seems to be related, as least in my case, to using the `hbs` fn from ember-template-imports in a component that also has a separate template file.

The error:

```
Build Error (broccoli-persistent-filter:Babel > [Babel: myapp]) in myapp/components/custom-table/index.js

/Users/foo/code/myapp/components/custom-table/index.js: Identifier 'hbs' has already been declared (10:9)

   9 | import { EMBER_CORE_TEAM } from from 'myapp/data/fixtures;
> 10 | import { hbs } from 'ember-template-imports';
     |          ^
```

---

### Reproduction

This works:

```js
// app/components/custom-table-component/index.js

import Component from '@glimmer/component';
import DataTable from 'myapp/components/data-table';
import Link from 'myapp/components/link';
import TruncateText from 'myapp/components/truncate-text';
import { EMBER_CORE_TEAM } from 'myapp/data/fixtures';
import { hbs } from 'ember-template-imports';

const formatSocialMediaHandle = (handle) => `@${handle}`;

const GitHubCell = hbs`
  <Link @href="https://github.com/{{@row.data.github}}">
    {{formatSocialMediaHandle @row.data.github}}
  </Link>
`;

const TwitterCell = hbs`
  <Link @href="https://twitter.com/{{@row.data.twitter}}">
    {{formatSocialMediaHandle @row.data.twitter}}
  </Link>
`;

export default class CustomTableComponent extends Component {
  columns = [
    { name: 'First name', valuePath: 'firstName' },
    { name: 'Last name', valuePath: 'lastName' },
    { name: 'GitHub', Cell: GitHubCell },
    { name: 'Twitter', Cell: TwitterCell },
  ];

  data = EMBER_CORE_TEAM.slice(0, 10);

  static template = hbs`
    <DataTable @columns={{this.columns}} @data={{this.data}} />
  `;
}
```

But does not work if I move the template for the `CustomTableComponent` into a separate template file:

```diff
// app/components/custom-table-component/index.js

export default class CustomTableComponent extends Component {
  columns = [
    { name: 'First name', valuePath: 'firstName' },
    { name: 'Last name', valuePath: 'lastName' },
    { name: 'GitHub', Cell: GitHubCell },
    { name: 'Twitter', Cell: TwitterCell },
  ];

  data = EMBER_CORE_TEAM.slice(0, 10);
-
-  static template = hbs`
-    <DataTable @columns={{this.columns}} @data={{this.data}} />
-  `;
}
```

```hbs
// app/components/custom-table-component/index.hbs

<DataTable @columns={{this.columns}} @data={{this.data}} />
```

**Comment by chriskrycho** [comment]
Looking at this—it looks like the issue is a bad/confusing error message? I wouldn't expect this to work, but it'd definitely be nice to improve the message here.

**Comment by HeroicEric** [comment]
@chriskrycho When you say "I wouldn't expect this to work", is that because of the implementation or because you don't think it _should_ work?

If you don't think it should work, why not? This seems like a valid use-case to me but maybe I'm missing something.

**Comment by chriskrycho** [comment]
Huh, I am not entirely sure what I meant back in August to be honest. On rereading it just now, I would *also* expect this to work, though the semantics are kind of weird (you ultimately end up with a single JS module with components in both strict and loose mode).

For anyone who is unfamiliar, colocated component templates use the same underlying mechanics as this library does: the build grabs the `.hbs` file and inlines it into a `precompileTemplate` call in the associated `.js` file, stitching it together with the backing class using `setComponentTemplate`. So the result of the code sample you've given should be, more or less, this—either way:

```js
// app/components/custom-table-component/index.js

import Component from '@glimmer/component';
import DataTable from 'myapp/components/data-table';
import Link from 'myapp/components/link';
import TruncateText from 'myapp/components/truncate-text';
import { EMBER_CORE_TEAM } from 'myapp/data/fixtures';
import { precompileTemplate } from '@ember/template-compilation';
import { setComponentTemplate } from '@ember/component';

const formatSocialMediaHandle = (handle) => `@${handle}`;

const GitHubCell = hbs`
  <Link @href="https://github.com/{{@row.data.github}}">
    {{formatSocialMediaHandle @row.data.github}}
  </Link>
`;

const TwitterCell = hbs`
  <Link @href="https://twitter.com/{{@row.data.twitter}}">
    {{formatSocialMediaHandle @row.data.twitter}}
  </Link>
`;

class CustomTableComponent extends Component {
  columns = [
    { name: 'First name', valuePath: 'firstName' },
    { name: 'Last name', valuePath: 'lastName' },
    { name: 'GitHub', Cell: GitHubCell },
    { name: 'Twitter', Cell: TwitterCell },
  ];

  data = EMBER_CORE_TEAM.slice(0, 10);
}

setComponentTemplate(
  precompileTemplate(`
    <DataTable @columns={{this.columns}} @data={{this.data}} />
  `),
  CustomTableComponent
);

export default CustomTableComponent;
```

Since scope isn’t involved, and since we don’t (to my knowledge, anyway) do anything to add `hbs` in colocated templates, this *should* just work. :thinking: Maybe we *do* add it for some reason? I’d have to go look at that part of the build.

@HeroicEric, do you have any more of the stack trace here? Also, does it reproduce when using the changes at #25?

**Comment by dfreeman** [comment]
I'm not 100% what would have led to this situation before, but I think there's a _decent_ chance it's fixed by #25, as `babel-import-util` is careful not to trample identifiers that are already in use.

**Comment by NullVoxPopuli** [comment]
Closing as hbs isn't supported anymore and has been removed in v4.
